salt/modules/useradd.py

```diff
@@ -306,98 +381,132 @@ def getent(refresh=False):
         return __context__['user.getent']
 
     ret = []
-    for data in pwd.getpwall():
+    if root is not None and __grains__['kernel'] != 'AIX':
+        _fun = functools.partial(_getpwall, root=root)
+    else:
+        _fun = functools.partial(pwd.getpwall)
+
+    for data in _fun():
         ret.append(_format_info(data))
     __context__['user.getent'] = ret
     return ret
 
 
-def chuid(name, uid):
+def _chattrib(name, key, value, param, persist=False, root=None):
+    '''
+    Change an attribute for a named user
+    '''
+    pre_info = info(name, root=root)
+    if not pre_info:
+        raise CommandExecutionError('User \'{0}\' does not exist'.format(name))
+
+    if value == pre_info[key]:
+        return True
+
+    cmd = ['usermod']
+
+    if root is not None and __grains__['kernel'] != 'AIX':
```

Comment:
You mean, this is fine on Windows, OSX and Solaris?

Reply:
nop, see `__virtual__`